body/src/main/java/com/networknt/body/BodyConverter.java

```diff
@@ -20,15 +20,25 @@
 
 import java.util.*;
 
+/**
+ * To convert the form into a map, we first convert the formValue into a String and then put into the map.
+ * For the same form key, there might be multiple form values, we handle it differently based on the size.
+ */
 public class BodyConverter {
     static Map<String, Object> convert(FormData data) {
         Map<String, Object> map = new HashMap<>();
         for (String key : data) {
-            List<Object> list = new ArrayList<>();
-            for (FormData.FormValue value : data.get(key)) {
-                list.add(value);
+            if (data.get(key).size() == 1) {
```

Comment:
looks good.
data.get(key) won't be null even send only a key without value based on @jiachen1120 testing.

Reply:
Thanks for the test confirmation.